src/backend/distributed/commands/multi_copy.c

```diff
@@ -2249,6 +2306,14 @@ CitusSendTupleToPlacements(TupleTableSlot *slot, CitusCopyDestReceiver *copyDest
 		}
 	}
 
+	if (copyDest->shouldUseLocalCopy && shardState->containsLocalPlacement)
+	{
+		bool isEndOfCopy = false;
+		ProcessLocalCopy(slot, copyDest, shardId, shardState->localCopyBuffer,
```

Comment:
This name is vague. Maybe WriteTupleToLocalShard? The fact that the function buffers writes and only sometimes writes to the table is mostly an implementation detail.

(I also wonder whether we should pass the whole CopyShardState to further hide that)